src/Microsoft.Framework.Caching.SqlServer/SqlQueries.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Framework.Caching.SqlServer
+{
+    internal class SqlQueries
+    {
+        private const string CreateTableFormat = "CREATE TABLE [{0}](" +
+            "Id nvarchar(100)  NOT NULL PRIMARY KEY, " +
+            "Value varbinary(MAX) NOT NULL, " +
+            "ExpiresAtTimeUTC datetimeoffset NOT NULL, " +
+            "SlidingExpirationInTicks bigint NULL," +
+            "AbsoluteExpiration datetimeoffset NULL)";
+
+        private const string CreateNonClusteredIndexOnExpirationTimeFormat
+            = "CREATE NONCLUSTERED INDEX Index_ExpiresAtTimeUTC ON [{0}](ExpiresAtTimeUTC)";
+
+        // Used for getting table schema when trying to 'Connect' to the database
+        private const string GetTableSchemaFormat = "SELECT * FROM [{0}]";
+
+        private const string GetCacheItemFormat =
+            "SELECT Id, Value, ExpiresAtTimeUTC, SlidingExpirationInTicks, AbsoluteExpiration " +
+            "FROM [{0}] WHERE Id = @Id AND @UtcNow <= ExpiresAtTimeUTC";
+
+        private const string AddCacheItemFormat = "INSERT INTO [{0}] " +
```

Comment:
Actually, thinking about it more, if we want to avoid race conditions, and get last-one-wins semantics without blocking we are probably going to need to do something smarter. Not sure yet what.